packages/jupyter-ai/src/components/chat-settings.tsx

```diff
@@ -29,6 +32,50 @@ enum ChatSettingsState {
   Success
 }
 
+function getProviderId(globalModelId: string | null) {
+  if (!globalModelId) {
+    return null;
+  }
+
+  return globalModelId.split(':')[0];
+}
+
+function getLocalModelId(globalModelId: string | null) {
+  if (!globalModelId) {
+    return null;
+  }
+
+  const components = globalModelId.split(':');
```

Comment:
Should we validate that len(components) == 2 or 1, to prevent an incorrect model ID like `foo:bar:baz` from being provided?

Reply:
The list of models returned from the ListProviders API should always return valid global model IDs. If we need further validation, this should be done in the backend instead.